src/preprocess/opis.py

```diff
@@ -22,6 +22,8 @@ def get_treatment_data(
     df = pd.read_parquet(f'{data_dir}/opis.parquet.gzip')
     df = filter_treatment_data(df, drugs, regimens)
     df = process_treatment_data(df)
+    # fix height values
+    df=df.groupby('mrn').apply(lambda group: group.assign(height=group['height'].max())).reset_index(drop=True)
```

Comment:
I think this would be more readable and efficient (you can check with %%timeit)

```
max_height_map = df.groupby('mrn')['height'].max()
df['height'] = df['mrn'].map(max_height_map)
```

Also, I would put this in process_treatment_data, and clarify in the comment that height values are a bit noisy so taking the maximum height values

Reply:
This can be improved even more actually

`df['height'] = df.groupby('mrn')['height'].transform('max')`